main.go

```diff
@@ -195,6 +195,8 @@ type BackendStats struct {
 	Tx              int64
 	UpSince         time.Time
 	DowntimeStart   time.Time
+	LastFinished    time.Time
+	CurrentCalls    int
```

Comment:
I think it is reasonable to make these atomic values. Use `int64` for both and `time.Time.UnixNano()` for timestamp.

If there are two updating the `LastFinished` at the same time it shouldn't really be a big issue.